### ISSUE TITLE
Missing documentation on "localize" option

### ISSUE THREAD

**Issue by DavidVotrubec** [nrwl/nx#4751]
## Current Behavior
Missing info on i18n localization in documentation. Also the existing documentation about i18n is very brief and without any examples. It seems not to be compatible with Angular localization configuration (in angular.json file).

Screenshot of missing info to "localize" property
![image](https://user-images.githubusercontent.com/454429/107496544-ba012800-6b91-11eb-86d4-e39b737932cc.png)

It seems to expect either boolan or array, but there is no example mentioned.
![image](https://user-images.githubusercontent.com/454429/107496754-00ef1d80-6b92-11eb-8bd1-0126aced36d4.png)

## Expected Behavior
Complete documentation with examples. Or recipe in cookbook.
Also it should be compatible with how i18n is handled in angular.json.

## Steps to Reproduce
see above

**Comment by PhilippeMorier** [comment]
It would be very helpful if there would be more documentation on how to officially setup a workspace with `@angular/localize`. Any updates on this?
Much appreciated!

I've only found
- https://blog.davidjs.com/2021/03/working-with-angular-i18n-inside-nrwl-nx-workspace/
- https://stackoverflow.com/questions/72728466/nx-monorepo-14-3-6-no-support-for-localize-angular-localize
- https://github.com/nrwl/nx/issues/4584

**Comment by mackelito** [comment]
I'm also looking for information regarding this :)

**Comment by PhilippeMorier** [comment]
I ended up doing these steps which are mainly following the steps described in the Angular docs :tada:.

1. `npm i -D @angular/localize`
2. Add `import '@angular/localize/init';` in `apps/<your-app>/src/polyfills.ts`
3. [Define locales in the build configuration](https://angular.io/guide/i18n-common-merge#define-locales-in-the-build-configuration) in `apps/<you-app>/project.json` at the "root" level. I.e. add as sibling of `prefix`, `targets` etc.:  
   ```json
   "i18n": {
     "sourceLocale": "en-US",
     "locales": {
       "en-GB": "apps/<your-app>/src/locale/messages.en-GB.xlf",
       "fr": "apps/<your-app>/src/locale/messages.fr.xlf"
     }
   },
   ```  
   This tells Angular that the language in your source files is `en-US` and you support the locales `en-GB` & `fr`. You don't have to create any of the files yet.
4. [Report missing translations](https://angular.io/guide/i18n-common-merge#report-missing-translations) in `apps/<your-app>/project.json`. I.e. add under `targets`➔`build`➔`options`:  
   ```json
   "i18nMissingTranslation": "error",
   "localize": ["en-US"]
   ```  
   This makes the build/serve command failing if there are missing translations for the supported locales. Furthermore, it defines `en-US` as the locale used if not specified when build/serve the app.
4. [Apply specific build options for just one locale](https://angular.io/guide/i18n-common-merge#apply-specific-build-options-for-just-one-locale) in `apps/<your-app>/project.json`
   1. Add under `targets`➔`build`➔`configurations`:  
      ```json
      "en-GB": {
        "localize": ["en-GB"]
      },
      "fr": {
        "localize": ["fr"]
      }
      ```
   1. Add under `targets`➔`serve`➔`configurations`:  
      ```json
      "en-GB": {
        "browserTarget": "field:build:development,en-GB"
      },
      "fr": {
        "browserTarget": "field:build:development,fr"
      }
      ```  
   This will allow you to be able to serve your app given a locale e.g.:
   `nx serve <your-app> --configuration=en-GB`
6. Again in `apps/<your-app>/project.json` under `targets`➔`extract-i18n`➔`options` add:  
   `"outputPath": "apps/<your-app>/src/locale"`
7. Run `nx extract-i18n <your-app>` to extract all your text "marked" with `i18n`/`$localize` in your components into `apps/<your-app>/src/locale/messages.xlf`
8. [Create a translation file for each language](https://angular.io/guide/i18n-common-translation-files#create-a-translation-file-for-each-language)
   1. Copy `messages.xlf`
   2. Rename copied file and add locale postfix, e.g.: `messages.fr.xlf`
   3. Add translations in `<target>` tag

Now you should be setup and able to run following commands :muscle::
* `nx extract-i18n <your-app>` to re-generate `messages.xlf` (e.g. after adding new translations)
* `nx serve <your-app>` to serve app with default locale
* `nx serve <your-app> --configuration=en-GB` to serve app with the given locale
* `nx build <your-app> --localize` to build one app per configured locale

If you are using Storybook we need to [make it aware of the localize package](https://stackoverflow.com/a/60414611/3731530) by either
* Adding `import '@angular/localize/init';` to `preview.js`  
or
* Adding `import '../src/polyfills';` to `preview.js` to import all polyfills

Updating/merging translation files is NOT natively supported by Angular:
* https://github.com/angular/angular-cli/issues/6552#issuecomment-316038368
* https://github.com/angular/angular/issues/37655#issuecomment-937731190

**Comment by sac-cas** [comment]
@PhilippeMorier: Thanks for sharing your configuration steps. How is your experience using angular’s i18n with nx workspace? did you use it in a real world project with a few angular apps running?
I wonder how big the overhead for the configuration is. With four apps and three languages you have twelve applications running.
In the next few days I will try i18next but I would like to see some support for configuration instead of copy & paste it to all my apps. There is not much described in the documentation.

**Comment by david-ru** [comment]
@PhilippeMorier thanks for sharing this. I've have tried them but the server response is: `Error: Configuration 'en-GB' is not set in the workspace.`. However, this works well, for both the remote and the host, if I just use `nx serve <my-app-name>`. 
My configuration is as follows:
```ts
"build": {
   "executor": "@nrwl/angular:webpack-browser",
   .....
   "configurations": {
        "en-GB": {
          "localize": ["en-GB"]
        },
        "es": {
          "localize": ["es"]
        },
   ....
   }
},
.......
"serve": {
      "executor": "@nrwl/angular:module-federation-dev-server",
      "configurations": {
        "en-GB": {
          "browserTarget": "host-app:build:development,en-GB"
        },
        "es": {
          "browserTarget": "host-app:build:development,es"
        },
.....
```

Any clue about what might be happening off the top of your head?

**Comment by PhilippeMorier** [comment]
Hey @david-ru!

1. Do both "language" files exist? I.e.:
   - `messages.en-GB.xlf` &
   - `messages.es.xlf`

2. What is your `i18n.sourcleLocale/locales`?  
   ```json
   "i18n": {
       "sourceLocale": "en-US",
       "locales": {
         "en-GB": "apps/host-app/src/locale/messages.en-GB.xlf",
         "es": "apps/host-app/src/locale/messages.es.xlf"
       }
     },
   ```

3. What is your  `targets.build.options.localize`?
   ```json
   {
     "targets": {
       "build": {
         "options": {
           "localize": ["en-US"]
         }
       }
     }
   }
   ```

**Comment by david-ru** [comment]
Thanks for answering ;-). So, my answers are:

1. Yes:
![Screenshot 2023-02-06 at 15 09 05](https://user-images.githubusercontent.com/22265948/216993362-bfed3378-b8da-42d0-9d84-5fa3225f8d27.png)

2. `i18n.sourceLocale/locales`: 
![Screenshot 2023-02-06 at 15 13 30](https://user-images.githubusercontent.com/22265948/216994337-2da9bf05-ca42-46cc-837f-3451e337ff8c.png)

3. `targets.build.options.localize`:
![Screenshot 2023-02-06 at 15 14 38](https://user-images.githubusercontent.com/22265948/216994383-1d24e95b-53bc-4058-8f6f-4e07a5785aab.png)

I include screenshots of the rest just in case:

4. `targets.build.configuration`:
![Screenshot 2023-02-06 at 15 21 53](https://user-images.githubusercontent.com/22265948/216996564-d717a754-81e2-49d6-810d-4d6a628eb597.png)

5. `serve.configurations`: 
![Screenshot 2023-02-06 at 15 22 18](https://user-images.githubusercontent.com/22265948/216996814-eddfc5c8-d5c9-4ea7-9389-c2595568782e.png)

**Comment by PhilippeMorier** [comment]
Just to be sure :innocent: ... 
1. the `messages.xlf` does exist?
2. what is your exact command you run which causes the error?

**Comment by david-ru** [comment]
Yeh, no problem :-)

1. I added that `messages.xlf` but still no luck. 
2. That's the one.
3. Exactly. I have a host and a remote, plus two libraries (with their own project.json each). I had already followed that configuration but... nothing. Tried to get rid of the `defaultConfiguration` also in both microfrontends and libraries, but no luck either.

The whole app is run using `nx serve host-app` and it automatically starts the remote and the libraries, so I guess this shouldn't be necessary `nx serve remote-app --configuration=en-GB`.

Next steps from my part could be to try run it having only the host and see what happens. I'll get back to you if I get something positive.
